pkg/controller/util.go

```diff
@@ -331,12 +332,58 @@ func GetScratchPvcStorageClass(client kubernetes.Interface, cdiclient clientset.
 	return ""
 }
 
+// GetPodResourceRequirements returns the ResourceRequirements needed to create temporary pods.
+// If the ResourceRequirements values are specified in CDIConfig, these values are taken and used.
+// If not, 0 is used as default values.
+func GetPodResourceRequirements(cdiclient clientset.Interface) (*v1.ResourceRequirements, error) {
+	config, err := cdiclient.CdiV1alpha1().CDIConfigs().Get(common.ConfigName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Unable to find CDI configuration, %v\n", err)
+		return nil, err
+	}
+
+	cpuLimit := resource.NewQuantity(0, resource.DecimalSI)
```

Comment:
Am I missing something here, can't you simply return config.Status.DefaultPodResourceRequirements (once that is added)?
If you don't want to return the same section of memory and want a copy you can do:
```go
if config.Spec.PodResourceRequirements != nil {
  return &config.Spec.PodResourceRequirements.DeepCopy(), nil
} else {
  return nil, nil
}
```

Reply:
We decided to use Status.DefaultPodResourceRequirements.Limits.Cpu(),  Status.DefaultPodResourceRequirements.Limits.Memory(), Status.DefaultPodResourceRequirements.Requests.Cpu(), Status.DefaultPodResourceRequirements.Requests.Memory() respectively
rather than to use Status.DefaultPodResourceRequirements.DeepCopy() directly.

Since someone could insert only limits.cpu in Spec.PodResourceRequirements, when the ResourceQuota requires both limits.cpu and limits.memory. Then importer pod won't create as we want.
Thus we thought we should nil-check for every PodResourceRequirements.XXX.XXX() components. Is there any idea that I'm missing?

Reply:
So, that is why we want you to use cdiconfig.status.PodResourceRequirements, since the status is generated by the config-controller (from the spec.PodResourceRequirements and default values encoded in the controller), it will always have all values set to something, and thus you can always copy it because it is guaranteed to always be fully populated.

Reply:
Ok i got it. Thanks :)